libs/usb-drive/src/usb_drive.ts

```diff
@@ -152,7 +153,100 @@ function isFat32(deviceInfo: BlockDeviceInfo): boolean {
   return deviceInfo.fstype === 'vfat' && deviceInfo.fsver === 'FAT32';
 }
 
-export function detectUsbDrive(): UsbDrive {
+async function logMountInit(logger: Logger): Promise<void> {
+  await logger.log(LogEventId.UsbDriveMountInit, 'system');
+}
+
+async function logMountSuccess(logger: Logger): Promise<void> {
+  await logger.log(LogEventId.UsbDriveMounted, 'system', {
+    disposition: 'success',
+    message: 'USB drive successfully mounted.',
+  });
+}
+
+async function logMountFailure(logger: Logger, error: Error): Promise<void> {
+  await logger.log(LogEventId.UsbDriveMounted, 'system', {
+    disposition: 'failure',
+    message: 'USB drive failed to mount.',
+    error: error.message,
+    result: 'USB drive not mounted.',
+  });
+}
+
+export async function logEjectInit(
+  logger: Logger,
+  loggingUserRole: LoggingUserRole
+): Promise<void> {
+  await logger.log(LogEventId.UsbDriveEjectInit, loggingUserRole);
+}
+
+export async function logEjectSuccess(
+  logger: Logger,
+  loggingUserRole: LoggingUserRole
+): Promise<void> {
+  await logger.log(LogEventId.UsbDriveEjected, loggingUserRole, {
+    disposition: 'success',
+    message: 'USB drive successfully ejected.',
+  });
+}
+
+async function logEjectFailure(
+  logger: Logger,
+  loggingUserRole: LoggingUserRole,
+  error: Error
+): Promise<void> {
+  await logger.log(LogEventId.UsbDriveEjected, loggingUserRole, {
+    disposition: 'failure',
+    message: 'USB drive failed to eject.',
+    error: error.message,
+    result: 'USB drive not ejected.',
+  });
+}
+
+async function logFormatInit(logger: Logger, loggingUserRole: LoggingUserRole) {
+  await logger.log(LogEventId.UsbDriveFormatInit, loggingUserRole);
+}
+
+async function logFormatSuccess(
+  logger: Logger,
+  loggingUserRole: LoggingUserRole,
+  volumeName: string
+) {
+  await logger.log(LogEventId.UsbDriveFormatted, loggingUserRole, {
+    disposition: 'success',
+    message: `USB drive successfully formatted with a single FAT32 volume named "${volumeName}".`,
+  });
+}
+
+async function logFormatFailure(
+  logger: Logger,
+  loggingUserRole: LoggingUserRole,
+  error: Error
+) {
+  await logger.log(LogEventId.UsbDriveFormatted, loggingUserRole, {
+    disposition: 'failure',
+    message: `Failed to format USB drive.`,
+    error: error.message,
+    result: 'USB drive not formatted, error shown to user.',
+  });
+}
+
+async function mount(
+  deviceInfo: BlockDeviceInfo,
+  logger: Logger
+): Promise<void> {
+  await logMountInit(logger);
+  try {
+    await mountUsbDrive(deviceInfo.path);
+    await logMountSuccess(logger);
+    debug('USB drive mounted successfully');
```

Comment:
@jonahkagan what was the reason you were using `console.log` before?

Reply:
Where was I using console.log? I don't remember any reason.

Reply:
In the `catch` clause the `void mount()` promise. I wondered if there was a need to use `console.log` when it's out of the normal execution like that. You even had a lint rule disabled for it 🤷

Reply:
Probably was just cutting corners for initial implementation. My guilty secret is that I always delay adding formal logging because I don't like how it clutters the code (as you can see from my other comments)

Reply:
I feel the same. I too often make decisions based off of code appearance.